subsys/bluetooth/host/hci_core.c

```diff
@@ -1377,6 +1377,8 @@ static void update_conn(struct bt_conn *conn, const bt_addr_le_t *id_addr,
 
 void bt_hci_le_enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 {
+	BT_ASSERT(evt->status == BT_HCI_ERR_SUCCESS);
```

Comment:
We should never assert (or rely on asserts) on parameters from 3rd party (we have to consider that the controller may be 3rd party here).